[TexMap] Add more tests for damage propagation

Scony opened the pull request:
#### 70183a2da5333bb8909fb7fcec7d8bd921939894
<pre>
[TexMap] Add more tests for damage propagation
<a href="https://bugs.webkit.org/show_bug.cgi?id=289332">https://bugs.webkit.org/show_bug.cgi?id=289332</a>

Reviewed by Carlos Garcia Campos.

This change adds layout tests covering the less common damage propagation scenarios.

* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/glib/damage/common.js:
* LayoutTests/platform/glib/damage/content/test.mp4: Added.
* LayoutTests/platform/glib/damage/image-as-layer-src-change-expected.txt: Added.
* LayoutTests/platform/glib/damage/image-as-layer-src-change.html: Added.
* LayoutTests/platform/glib/damage/image-src-change-expected.txt: Added.
* LayoutTests/platform/glib/damage/image-src-change.html: Added.
* LayoutTests/platform/glib/damage/layer-backdrop-filter-expected.txt: Added.
* LayoutTests/platform/glib/damage/layer-backdrop-filter.html: Added.
* LayoutTests/platform/glib/damage/layer-color-change-expected.txt: Added.
* LayoutTests/platform/glib/damage/layer-color-change.html: Added.
* LayoutTests/platform/glib/damage/layer-opacity-change-expected.txt: Added.
* LayoutTests/platform/glib/damage/layer-opacity-change.html: Added.
* LayoutTests/platform/glib/damage/layer-overlaps-with-opacity-expected.txt: Added.
* LayoutTests/platform/glib/damage/layer-overlaps-with-opacity.html: Added.
* LayoutTests/platform/glib/damage/layer-text-change-expected.txt: Added.
* LayoutTests/platform/glib/damage/layer-text-change.html: Added.
* LayoutTests/platform/glib/damage/layer-visibility-change-expected.txt: Added.
* LayoutTests/platform/glib/damage/layer-visibility-change.html: Added.
* LayoutTests/platform/glib/damage/video-expected.txt: Added.
* LayoutTests/platform/glib/damage/video.html: Added.

Canonical link: <a href="https://commits.webkit.org/292262@main">https://commits.webkit.org/292262@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/06fd37f33d942adbba351e799baca8d9fed59e47

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/95424 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/15027 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/4894 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/100474 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/45930 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/15314 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/23460 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/72784 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/30050 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/98427 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/11459 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/86145 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/53115 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/11171 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/3875 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/45266 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/81351 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/3970 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/102511 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/22474 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/16410 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/81800 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/22723 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/82154 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/81171 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20324 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/25747 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/3173 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/15784 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/22443 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/22102 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/25577 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/23842 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

spenap commented on LayoutTests/platform/glib/damage/image-as-layer-src-change.html:
```diff
@@ -0,0 +1,66 @@
+                  resolve(URL.createObjectURL(blob));
+              });
+          });
+      }
```
There are two tests, for now, that use `takeCanvasSnapshotAsBlobURL()`. It's almost justified if we were to move it into `common.js`, I think.

In the method, the canvas is constant,  I think, and doesn't feel like it needs to be inside the promise. The promise itself can be made less verbose -- but thatś probably a matter of personal preference. I don't think you need to specify `async` inside again

```suggestion
      async function takeCanvasSnapshotAsBlobURL() {
          const canvas = document.getElementsByTagName("canvas")[0];
          return new Promise(resolve =>
              canvas.toBlob(blob => resolve(URL.createObjectURL(blob)))
          );
      }
```

spenap commented on LayoutTests/platform/glib/damage/image-src-change.html:
```diff
@@ -0,0 +1,64 @@
+    </style>
+    <script src="./common.js"></script>
+    <script>
+      async function takeCanvasSnapshotAsBlobURL() {
```
then both tests would use the updated method

spenap reviewed (commented):
Minor comment regarding the blob function. I tried it from web inspector and seemed to be functionally equivalent to yours.

Scony committed:
[TexMap] Add more tests for damage propagation
https://bugs.webkit.org/show_bug.cgi?id=289332

Reviewed by Carlos Garcia Campos.

This change adds layout tests covering the less common damage propagation scenarios.

* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/glib/damage/common.js:
* LayoutTests/platform/glib/damage/content/test.mp4: Added.
* LayoutTests/platform/glib/damage/image-as-layer-src-change-expected.txt: Added.
* LayoutTests/platform/glib/damage/image-as-layer-src-change.html: Added.
* LayoutTests/platform/glib/damage/image-src-change-expected.txt: Added.
* LayoutTests/platform/glib/damage/image-src-change.html: Added.
* LayoutTests/platform/glib/damage/layer-backdrop-filter-expected.txt: Added.
* LayoutTests/platform/glib/damage/layer-backdrop-filter.html: Added.
* LayoutTests/platform/glib/damage/layer-color-change-expected.txt: Added.
* LayoutTests/platform/glib/damage/layer-color-change.html: Added.
* LayoutTests/platform/glib/damage/layer-opacity-change-expected.txt: Added.
* LayoutTests/platform/glib/damage/layer-opacity-change.html: Added.
* LayoutTests/platform/glib/damage/layer-overlaps-with-opacity-expected.txt: Added.
* LayoutTests/platform/glib/damage/layer-overlaps-with-opacity.html: Added.
* LayoutTests/platform/glib/damage/layer-text-change-expected.txt: Added.
* LayoutTests/platform/glib/damage/layer-text-change.html: Added.
* LayoutTests/platform/glib/damage/layer-visibility-change-expected.txt: Added.
* LayoutTests/platform/glib/damage/layer-visibility-change.html: Added.
* LayoutTests/platform/glib/damage/video-expected.txt: Added.
* LayoutTests/platform/glib/damage/video.html: Added.

Canonical link: https://commits.webkit.org/292262@main
